internal/providers/terraform/azure/vpn_gateway.go

```diff
@@ -0,0 +1,171 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMVpnGatewayRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_virtual_network_gateway",
+		RFunc: NewAzureRMVpnGateway,
+		Notes: []string{},
+	}
+}
+
+func NewAzureRMVpnGateway(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	var tunnel, connection, dataTransfers *decimal.Decimal
+	sku := "Basic"
+	region := lookupRegion(d, []string{})
+	zone := regionToZone(region)
+
+	if d.Get("sku").Type != gjson.Null {
+		sku = d.Get("sku").String()
+	}
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	if sku == "Basic" {
+		sku = "Basic Gateway"
+	}
+
+	costComponents = append(costComponents, vpnGateway(region, sku))
+
+	if u != nil && u.Get("s2s_tunnel").Type != gjson.Null {
+		tunnel = decimalPtr(decimal.NewFromInt(u.Get("s2s_tunnel").Int()))
+		if tunnel != nil {
+			tunnelLimits := []int{10, 30}
```

Comment:
I know there is an upper limit of 30 on the pricing page but I think it would be better to ignore it here.  As written if a user set s2s_tunnel: 45, they would get prices for 30 connections.  I think it would be better for them to get the price for 35, even though that supposedly isn't possible according to the pricing page.
```suggestion
			tunnelLimits := []int{10}
```